code/datums/soundOutput.dm

```diff
@@ -33,29 +72,54 @@
 		S.channel = T.channel
 	S.frequency = T.frequency
 	S.falloff = T.falloff
-	S.status = T.status
-	if(T.x && T.y && T.z)
-		var/turf/owner_turf = get_turf(owner.mob)
-		if(owner_turf)
-			// We're in an interior and sound came from outside
-			if(SSinterior.in_interior(owner_turf) && owner_turf.z != T.z)
-				var/datum/interior/VI = SSinterior.get_interior_by_coords(owner_turf.x, owner_turf.y, owner_turf.z)
-				if(VI && VI.exterior)
-					var/turf/candidate = get_turf(VI.exterior)
-					if(candidate.z != T.z)
-						return // Invalid location
-					S.falloff /= 2
-					owner_turf = candidate
-			S.x = T.x - owner_turf.x
-			S.y = 0
-			S.z = T.y - owner_turf.y
-		S.y += T.y_s_offset
-		S.x += T.x_s_offset
-		S.echo = SOUND_ECHO_REVERB_ON //enable environment reverb for positional sounds
+	S.status = update ? SOUND_UPDATE : T.status 
+	if(!update)
+		S.params = list("on-end" = ".soundend [S.channel] [T.source]")
+
+
+	var/turf/source_turf
+	if(!QDELETED(T.source))
+		source_turf = get_turf(T.source)
+
+		if(!update && istype(T.source, /atom/movable))
+			source_sounds[T.source] = T
+			RegisterSignal(T.source, COMSIG_MOVABLE_MOVED, PROC_REF(update_sounds))
+	else
+		source_turf = locate(T.x, T.y, T.z)
+
+	var/turf/owner_turf = get_turf(owner.mob)
+	if(owner_turf)
+		// We're in an interior and sound came from outside
+		if(SSinterior.in_interior(owner_turf) && owner_turf.z != T.z)
+			var/datum/interior/VI = SSinterior.get_interior_by_coords(owner_turf.x, owner_turf.y, owner_turf.z)
+			if(VI && VI.exterior)
+				var/turf/candidate = get_turf(VI.exterior)
+				if(candidate.z != T.z)
+					return // Invalid location
+				S.falloff /= 2
+				owner_turf = candidate
+		S.x = source_turf.x - owner_turf.x + offset_x
+		S.y = 0
+		S.z = source_turf.y - owner_turf.y + offset_y
+	S.y += T.y_s_offset
+	S.x += T.x_s_offset
+	S.echo = SOUND_ECHO_REVERB_ON
```

Comment:
Reverb should only be enabled for a sound that exists "in-world", i.e. is a positional sound and not an interface sound like an admin ping.

Previously the entire block from `var/turf/owner_turf ...` to `S.echo ...` was wrapped in `if(T.x && T.y && T.z)` as the test for being positional, in your case something like `if(source_turf.x && source_turf.y && source_turf.z)` may be more appropriate.